pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginsDirectoryVisitor.java

```diff
@@ -0,0 +1,45 @@
+package org.apache.pinot.spi.plugin;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Collection;
+import java.util.HashSet;
+
+
+class PluginsDirectoryVisitor extends SimpleFileVisitor<Path> {
+
+  private final Collection<Path> _pluginDirs = new HashSet<>();
+
+  private final PinotPluginHandler _pinotPluginHandler;
+
+  private final ShadedPluginHandler _shadedPluginHandler;
+
+  public PluginsDirectoryVisitor(PinotPluginHandler _pinotPluginHandler, ShadedPluginHandler _shadedPluginHandler) {
+    this._pinotPluginHandler = _pinotPluginHandler;
+    this._shadedPluginHandler = _shadedPluginHandler;
+  }
```

Comment:
the code style checker is going to complain about that. Attributes must be prefixed with `_` and variables must not be prefixed with `_`. Also `this.attribute` is forbidden.

I hate that code style but... one gets used to it :D